### DISCUSSION TITLE
Review request: Switching to Group in some pyparsing code

### DISCUSSION THREAD

**Discussion by aucampia** [pyparsing/pyparsing#449] [Show and tell]
Hi, I'm not that knowledgeable with pyparsing and would like a second opinion on this PR which replaces some custom class with Group

- https://github.com/RDFLib/rdflib/pull/2142

If anyone could take a brief look I would appreciate it.

**Comment by ptmcg** [comment]
I assume you are using the latest version of pyparsing (3.0.9)?

**Reply by aucampia** [reply]
We have no specific fixed version, we use the latest in CI, and I think we somewhat encourage our users to do the same. Everything worked fine with `pyparsing<3.0` though.

**Comment by ptmcg** [comment]
After cloning the repo, I added this to the bottom of parser.py:

```python
if __name__ == '__main__':
    import pyparsing as pp

    pp.autoname_elements()
    QueryUnit.create_diagram("rdflib_query_unit.html")
```
which generated this diagram:

[rdflib_query_unit.pdf](https://github.com/pyparsing/pyparsing/files/10050378/rdflib_query_unit.pdf)

Maybe this will give some insights into your parser.

**Reply by aucampia** [reply]
Thanks @ptmcg I will have a look at this
